### PR TITLE
S3 lock for core-vpc-sandbox

### PR DESCRIPTION
**Pull request by SteveLinden**
Initial set up for just core-vpc-sandbox for testing.

### CODE REVIEW

File: terraform/environments/core-vpc/iam.tf [hunk 1]
```diff
@@ -1,3 +1,48 @@
...
+  path        = "/"
+  description = "S3 and dynamo policy"
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : "s3:ListBucket",
+        "Resource" : "arn:aws:s3:::mybucket"
```
Comment (dms1981) [inline terraform/environments/core-vpc/iam.tf:16]: Probably want to change this to refer to the correct bucket.
Have you considered maybe using the output of a data call to populate this, to keep you from needing a hard coded reference in the policy?

File: terraform/environments/core-vpc/iam.tf [hunk 2]
```diff
@@ -1,3 +1,48 @@
...
+          "dynamodb:GetItem",
+          "dynamodb:PutItem",
+          "dynamodb:Query",
+          "dynamodb:UpdateItem"
+        ],
+        "Resource" : ["arn:aws:dynamodb:eu-west-2:${local.environment_management.modernisation_platform_account_id}:table/modernisation-platform-terraform-state-lock"],
+        "Condition" : {
+          "ForAllValues:StringEquals" : {
+            "dynamodb:LeadingKeys" : [
+              "myorg-terraform-states/myapp/production/tfstate",    // during a state lock the full state file is stored with this key
```
Comment (dms1981) [inline terraform/environments/core-vpc/iam.tf:39]: Are these key values correct? I suspect we're not using `myorg-terraform-states/myapp` in the DynamoDB config.

File: terraform/environments/core-vpc/iam.tf (hunk 1 above)
Comment (SteveLinden) [reply terraform/environments/core-vpc/iam.tf:16]: Bucket name corrected.

File: terraform/environments/core-vpc/iam.tf (hunk 2 above)
Comment (SteveLinden) [reply terraform/environments/core-vpc/iam.tf:39]: May be corrected with "modernisation-platform-terraform-state/global-resources/terraform.tfstate" and "modernisation-platform-terraform-state/global-resources/terraform.tfstate-md5"

Comment (SteveLinden) [reply terraform/environments/core-vpc/iam.tf]: Change made and pushed

**Review by julialawrence — APPROVED** [review]
LGTM. State locking is happening. Plans show no changes.

**Pull request merged**